sky/backends/backend_utils.py

```diff
@@ -802,7 +806,17 @@ def write_cluster_config(
     ]
 
     assert cluster_name is not None
-    credentials = sky_check.get_cloud_credential_file_mounts()
+    excluded_clouds = []
+    remote_authentication = skypilot_config.get_nested(
+        (str(cloud).lower(), 'remote_authentication'), 'USER_ACCOUNT')
+    if remote_authentication == 'SERVICE_ACCOUNT':
+        if cloud.require_credential_on_remote():
+            raise exceptions.InvalidConfigs(
+                'remote_authentication: SERVICE_ACCOUNT is specified in '
+                f'{skypilot_config.loaded_config_path!r} for {cloud}, but it '
+                'not supported this cloud, please use \'USER\' instead.')
+        excluded_clouds = [cloud]
+    credentials = sky_check.get_cloud_credential_file_mounts(excluded_clouds)
```

Comment:
If it's always a singleton, any reason to make the arg a list?

Reply:
It was for the generality of the function, as in the future it might be able to allow the remote instance to not have multiple clouds' credential, e.g., if the image has the cloud credential set already. I am ok to make it singleton first, but I guess it does not matter that much to have an argument as a list. : )